types/ember-data/index.d.ts

```diff
@@ -1950,47 +1952,47 @@ export namespace DS {
      * used when `boolean` is passed as the type parameter to the
      * [DS.attr](../../data#method_attr) function.
      */
-    class BooleanTransform extends Transform {}
+    class BooleanTransform extends Transform<boolean> {}
     /**
      * The `DS.DateTransform` class is used to serialize and deserialize
      * date attributes on Ember Data record objects. This transform is used
      * when `date` is passed as the type parameter to the
      * [DS.attr](../../data#method_attr) function. It uses the [`ISO 8601`](https://en.wikipedia.org/wiki/ISO_8601)
      * standard.
      */
-    class DateTransform extends Transform {}
+    class DateTransform extends Transform<Date> {}
     /**
      * The `DS.NumberTransform` class is used to serialize and deserialize
      * numeric attributes on Ember Data record objects. This transform is
      * used when `number` is passed as the type parameter to the
      * [DS.attr](../../data#method_attr) function.
      */
-    class NumberTransform extends Transform {}
+    class NumberTransform extends Transform<number> {}
     /**
      * The `DS.StringTransform` class is used to serialize and deserialize
      * string attributes on Ember Data record objects. This transform is
      * used when `string` is passed as the type parameter to the
      * [DS.attr](../../data#method_attr) function.
      */
-    class StringTransform extends Transform {}
+    class StringTransform extends Transform<string> {}
     /**
      * The `DS.Transform` class is used to serialize and deserialize model
      * attributes when they are saved or loaded from an
      * adapter. Subclassing `DS.Transform` is useful for creating custom
      * attributes. All subclasses of `DS.Transform` must implement a
      * `serialize` and a `deserialize` method.
      */
-    class Transform extends Ember.Object {
+    class Transform<Deserialized = any, Serialized = any> extends Ember.Object {
```

Comment:
These are optional, but are useful for specifying the return type for the above built-in transforms.

Reply:
I kind of want to make this:
```suggestion
    class Transform<Deserialized = unknown, Serialized = unknown> extends Ember.Object {
```
but worry that will break too many things for people.

Reply:
Yeah, I think that's correct, but may be problematic on the grounds you mentioned. We happen to use unknown on our app, but not everyone does https://github.com/CenterForOpenScience/ember-osf-web/blob/develop/app/transforms/array.ts.